### ISSUE TITLE
Support AbortSignal.timeout

### ISSUE THREAD

**Issue by johngeorgewright** [mo/abortcontroller-polyfill#73]
See https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout

> The static AbortSignal.timeout() method returns an [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) that will automatically abort after a specified time.

**Comment by ccxdev** [comment]
Important thing, because it's feature breaks safari < 16

**Comment by elhardoum** [comment]
Here's a temporary fix:
```js
// safari polyfill
if ("AbortSignal" in window) {
  AbortSignal.timeout = AbortSignal.timeout || ((duration) =>
  {
    const controller = new AbortController;
    setTimeout(() => controller.abort(), duration);
    return controller.signal;
  });
}
```

**Comment by DevSide** [comment]
@elhardoum This cannot work because your setTimeout will start when AbortSignal.timeout() is called instead of the request itself.

**Comment by TechQuery** [comment]
@mo @elhardoum Another static method `AbortSignal.any()` comes out: https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/any_static

**Comment by mo** [comment]
the timeout implemented by @TechQuery is available in 1.7.6 now
